### PR TITLE
Report issues #40

### PR DESCRIPTION
**Pull request by cmeesters**
to be considered as first draft to work upon

### CODE REVIEW

**Comment by cmeesters** [comment]
@lukashellmann fixed the typo noted yesterday.

Please be aware: The typesetting of `Snakemake` will change anyway. The PR for the layout depends too much on this and other PRs, hence it is waiting in the layout branch (which apparently I have not pushed, yet ...)

File: slides/common/Contributing.tex [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  	\begin{column}{.5\textwidth}
+  	  \tikzset{
+  	  	LabelStyle/.style = { rectangle, rounded corners, draw,
+  	  		minimum width = 2em, fill = yellow!50,
+  	  		text = red, font = \bfseries },
+  	  	VertexStyle/.append style = { inner sep=5pt,
+  	  		font = \Large\bfseries},
+  	  	EdgeStyle/.append style = {->, bend left} }
+    	\resizebox{\columnwidth}{!}{
+  	   \begin{tikzpicture}
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:78]: I don't like the figure. Should it not be the other way round? Several causes (programm, workflow, computer) pointing towards the issue.
Fine as a sketch, but should be replaced it the future

File: slides/common/Contributing.tex [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  	   	%\tikzset{EdgeStyle/.append style = {bend left = 50}}
+  	   	\Edge[label = 4,style={<-,bend left}](Workflow)(Program)
+  	   	\Edge[label = 5,style={<-,bend left}](Computer)(Workflow)
+  	   	%\Loop[dist = 4cm, dir = SOWE, label = 6](Program.west)
+  	   \end{tikzpicture}}
+  	\end{column}
+  \end{columns}
+  \pause
+  \begin{hint}
+  	It is important to figure out the underlying issue before reporting anything!
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:102]: "Please investigate the underlying issue before reporting an issue!"

thinking about it: many (biology) users will struggle doing this properly. Should we add a section on "how to identity the issue"?

File: slides/common/Contributing.tex [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  \end{hint}	
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+  \frametitle{Do I need a \texttt{GitHub} account?}
+  It may change, but today many software projects are either hosted on some institutional \texttt{GitLab} or the global git server of Microsoft, \texttt{GitHub}.\newline
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:109]: <s>It may change, but</s>
I would not mention MS

File: slides/common/Contributing.tex [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  If you want to report an issue or even contribute back to the project with anything else, you will need an account. \lhref{This page}{https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account} describe how.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:110]: this slide is very wordy. Maybe add linked logos of gitlab & github instead of the first sentence. full link for github registration is not necessary (i think a general github link is sufficient).

File: slides/common/Contributing.tex [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  \frametitle{Do I need a \texttt{GitHub} account?}
+  It may change, but today many software projects are either hosted on some institutional \texttt{GitLab} or the global git server of Microsoft, \texttt{GitHub}.\newline
+  If you want to report an issue or even contribute back to the project with anything else, you will need an account. \lhref{This page}{https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account} describe how. 
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Cluster Issues}
+	\begin{question}[Where to?]
+	  Report them to your cluster admins (chat, mail ticket, etc.)
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:117]: add icons: mail, messenger, speech bubble, ...

File: slides/common/Contributing.tex [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+	\end{question}
+    \pause
+    \begin{question}[What?]
+      Inaccessible paths (file system issue), network issues, node failures, etc.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:121]: Inaccessible paths, file system issue, network issues, node failures, etc

File: slides/common/Contributing.tex [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+	\begin{question}[Where to?]
+	  Report them to your cluster admins (chat, mail ticket, etc.)
+	\end{question}
+    \pause
+    \begin{question}[What?]
+      Inaccessible paths (file system issue), network issues, node failures, etc. 
+    \end{question}
+    \pause
+    \begin{question}[How?]
+      Give a meaningful brief statement. "I can't do \altverb{ls} on \altverb{/some/path}." is more meaningful than "The filesystem stopped working!" and "For me too!" messages.\newline
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:125]: <s>brief</s>
<s>and "For me too!"</s>

File: slides/common/Contributing.tex [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+      \bcattention Check your mails! Has some maintenance been scheduled?
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:126]: I would leave this out, even though I think this is often the cause ;-)

File: slides/common/Contributing.tex [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+    \end{question}	
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Application Issues}
+	\begin{question}[Where to?]
+		The issue tracker of your application (if any).
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:134]: icon: github etc?

File: slides/common/Contributing.tex [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Application Issues}
+	\begin{question}[Where to?]
+		The issue tracker of your application (if any). 
+	\end{question}
+	\pause
+	\begin{question}[What?]
+		Any application specific error. Try running the application with the input and parameters specified in the workflow (get them with \altverb{snakemake -p} to print the shell command or other debug flags). If the application breaks, report the failure. Else, it might be the workflow.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:138]: "Try running the application with the input and parameters specified in the workflow (get them with \altverb{snakemake -p} to print the shell command or other debug flags). If the application breaks, report the failure. Else, it might be the workflow."
this is too wordy. is it worth to creat a new slide? maybe a small flow chart instead?

File: slides/common/Contributing.tex [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+\begin{frame}
+  \frametitle{Do I need a \texttt{GitHub} account?}
+  It may change, but today many software projects are either hosted on some institutional \texttt{GitLab} or the global git server of Microsoft, \texttt{GitHub}.\newline
+  If you want to report an issue or even contribute back to the project with anything else, you will need an account. \lhref{This page}{https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account} describe how. 
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Cluster Issues}
+	\begin{question}[Where to?]
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:116]: <s>to</s>

File: slides/common/Contributing.tex [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+    \begin{question}[How?]
+      Give a meaningful brief statement. "I can't do \altverb{ls} on \altverb{/some/path}." is more meaningful than "The filesystem stopped working!" and "For me too!" messages.\newline
+      \bcattention Check your mails! Has some maintenance been scheduled?
+    \end{question}	
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Application Issues}
+	\begin{question}[Where to?]
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:133]: <s>to</s>

File: slides/common/Contributing.tex [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+	\pause
+	\begin{question}[How?]
+		Indicate the error message and what is leading to it. If you need special input to reproduce the issue, upload the input somewhere or attach minimal input to the issue report.
+	\end{question}
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Workflow Issues}
+	\begin{question}[Where to?]
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:149]: <s>to</s>

File: slides/common/Contributing.tex [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+		The issue tracker of your application (if any). 
+	\end{question}
+	\pause
+	\begin{question}[What?]
+		Any application specific error. Try running the application with the input and parameters specified in the workflow (get them with \altverb{snakemake -p} to print the shell command or other debug flags). If the application breaks, report the failure. Else, it might be the workflow.
+	\end{question}
+	\pause
+	\begin{question}[How?]
+		Indicate the error message and what is leading to it. If you need special input to reproduce the issue, upload the input somewhere or attach minimal input to the issue report.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:142]: "Provide the error message. It's often helpful to provide (sample) input to the developer to reproduce the issue."

File: slides/common/Contributing.tex [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+	\frametitle{Reporting Workflow Issues}
+	\begin{question}[Where to?]
+		The issue tracker at your workflow. For \texttt{Snakemake} somewhere in the \lhref{https://snakemake.github.io/snakemake-workflow-catalog/}{catalog}. 
+	\end{question}
+	\pause
+	\begin{question}[What?]
+	   Check that it is you causing the error, e.\,g. undefined variables. \texttt{Snakemake} will inform you about the nature of the issue. Read the error output carefully. Report when sure it is not you.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:154]: replace "Read the error output carefully. Report when sure it is not you." with "Read the error message and check input and parameters. Report an issue"

The phrase "when sure it is not you" sounds judgmental.

File: slides/common/Contributing.tex [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+	\end{question}
+	\pause
+	\begin{question}[How?]
+		Indicate the error message and what is leading to it. If you need special input to reproduce the issue, upload the input somewhere or attach minimal input to the issue report.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:158]: "Provide the error message. It's helpful to provide (sample) input to the developer to reproduce the issue."

File: slides/common/Contributing.tex [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+	\end{question}
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+  \frametitle{Reporting \texttt{Snakemake} Issues}
+  \begin{question}[Where to?]
+  	 The \lhref{https://github.com/snakemake/snakemake/issues}{issue tracker of the \texttt{Snakemake} project}.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:166]: <s>to<s/>

File: slides/common/Contributing.tex [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+  \end{question}	
+  \pause
+  \begin{question}[What?]
+  	Clearly unexpected behaviour. Or tracebacks pointing to \texttt{Snakemake} itself.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:170]: "Unexpected behavior and tracebacks pointing to \texttt{Snakemake} itself"

File: slides/common/Contributing.tex [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+  \begin{question}[Where to?]
+  	 The \lhref{https://github.com/snakemake/snakemake/issues}{issue tracker of the \texttt{Snakemake} project}.
+  \end{question}	
+  \pause
+  \begin{question}[What?]
+  	Clearly unexpected behaviour. Or tracebacks pointing to \texttt{Snakemake} itself. 
+  \end{question}
+  \pause
+  \begin{question}[How?]
+  	Indicate the error message and what is leading to it (e.\,g. the command line you used). If you need special input to reproduce the issue, upload the input somewhere or attach minimal input to the issue report.
```
Comment (lukashellmann) [inline slides/common/Contributing.tex:174]: "Provide the error message. It's helpful to provide (sample) input and command line used to the developer to reproduce the issue."

**Review by lukashellmann — APPROVED** [review]
some points might be worth additional slides (see comments). while the structure is fine, we should make the slides more visually appealing in the future

File: slides/common/Contributing.tex [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+	\end{docs}
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{frame}
+  \frametitle{"I am not a Programmer!"}
+  \ldots and do not know what and how to contribute!
+  \pause
+  \begin{block}{Open Source is \emph{NOT JUST} about Programming}
+  	\lhref{https://opensource.guide/how-to-contribute/}{You can}
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply slides/common/Contributing.tex:41]: well spotted. Thank you.

done.

File: slides/common/Contributing.tex [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+  	\begin{itemize}[<+->]
+  		\item contribute by writing whats wrong (issue reports)
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply slides/common/Contributing.tex:43]: done

File: slides/common/Contributing.tex [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+  		\item contribute by writing how it's done (documentation)
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply slides/common/Contributing.tex:44]: done

File: slides/common/Contributing.tex [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+  		\item contribute by organizing events (bringing people together)
+  		\item contribute by checking (test the code submissions of other's)
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply slides/common/Contributing.tex:46]: done

File: slides/common/Contributing.tex [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
+	\frametitle{What is this about?}
+	\begin{question}[Questions]
+		\begin{itemize}
+			\item How do I report Issues?
+			\item How do I improve the Documentation?
+		\end{itemize}
+	\end{question}
+	\begin{docs}[Objectives]
+		\begin{enumerate}
+			\item Getting to know \texttt{GitHub}.
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply slides/common/Contributing.tex:28]: done

File: slides/common/Contributing.tex (hunk 1 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:78]: well spotted. Changed arrow direction accordingly.

Note, that while I always appreciate better sketches/images/etc. I am well aware not to be the best designer of such things ...

File: slides/common/Contributing.tex (hunk 2 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done - albeit in an extra text-only slide.

File: slides/common/Contributing.tex (hunk 3 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:120]: done

File: slides/common/Contributing.tex (hunk 4 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:121]: You are right. This most mostly due to confused syntax. Still, the text has been shortened and an extra box introduced for a higher visual appeal.

File: slides/common/Contributing.tex (hunk 5 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 7 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 8 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: hm, the mail-check remark is there for the relief of colleagues. You have no idea ;-).

File: slides/common/Contributing.tex (hunk 9 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 10 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: please take a look - a tikz flow chart has been added - intend to do this for the next 2 slides, too.

File: slides/common/Contributing.tex (hunk 11 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 12 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 13 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: done

File: slides/common/Contributing.tex (hunk 14 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: much better! Done.

File: slides/common/Contributing.tex (hunk 15 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:213]: right. done. Please check.

File: slides/common/Contributing.tex (hunk 16 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:217]: duplicate, done

File: slides/common/Contributing.tex (hunk 17 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:225]: done

File: slides/common/Contributing.tex (hunk 18 above)
Comment (cmeesters) [reply slides/common/Contributing.tex:229]: done

File: slides/common/Contributing.tex (hunk 19 above)
Comment (cmeesters) [reply slides/common/Contributing.tex]: duplicate. done.

**Comment by mptrsen** [comment]
all points resolved, can be merged?

**Comment by cmeesters** [comment]
@lukashellmann just merged latest main. Could you take a look well before the upcoming course - too much needs to be done, anyways. Thank you.

**Comment by cmeesters** [comment]
@mptrsen I think so, yes.
